beacon_node/beacon_chain/src/custody_context.rs

```diff
@@ -150,6 +150,24 @@ impl ValidatorRegistrations {
                 .or_insert(latest_validator_custody);
         }
     }
+
+    /// Updates the `epoch -> cgc` map by pruning records before `effective_epoch`
+    /// while setting the `cgc` at `effective_epoch` to the latest validator custody requirement.
+    ///
+    /// This is used to restart custody backfill sync at `effective_epoch`
+    pub fn reset_validator_custody_requirements(&mut self, effective_epoch: Epoch) {
+        if let Some(latest_validator_custody_requirements) =
+            self.latest_validator_custody_requirement()
+        {
+            self.epoch_validator_custody_requirements
+                .retain(|&epoch, _| epoch >= effective_epoch);
+
+            self.epoch_validator_custody_requirements
+                .entry(effective_epoch)
```

Comment:
shouldn't this be `effective_epoch + 1`

We don't have the new CGC until backfill for  the epoch is completed right?

Reply:
I think you might be mixing this up with the `earliest_data_column_slot`? It seems that the `epoch` values in this map mean:

- for epochs `>= epoch`, the custody requirement is X.

So using `epoch` rather than `epoch + 1` seems right to me?